Tests/anvil-csharp-tests.asmdef

```diff
@@ -0,0 +1,16 @@
+{
+    "name": "anvil-csharp-tests",
+    "rootNamespace": "Anvil.CSharp.Tests",
+    "references": [],
+    "includePlatforms": [
+        "Editor"
+    ],
+    "excludePlatforms": [],
+    "allowUnsafeCode": false,
+    "overrideReferences": false,
+    "precompiledReferences": [],
+    "autoReferenced": true,
+    "defineConstraints": [],
+    "versionDefines": [],
+    "noEngineReferences": false
```

Comment:
We probably want to exclude Unity engine references for any pure c-sharp libs

Reply:
There's already an `anvil-csharp-core.asmdef` in the root of the repo for the same purpose - it's meaningless and harmless in non-Unity projects, and prevents there from being setup steps before Anvil is functional in Unity projects.

Reply:
Ok sound good!